api/providers/AddressProvider.cs

```diff
@@ -53,4 +54,23 @@ public List<Address> GetAll()
     {
         throw new NotImplementedException();
     }
+
+    public Address? GetOrCreateAddress(SupplierRequest request)
+    {
+
+        if (request.Address_id != null)
```

Comment:
```suggestion
        if (request.AddressId!= null)
```

Graag PascalCase aanhouden. (en vranderen in de AddressDTO)

Reply:
addressId bestaat niet in de addressdto